File: mobile/lib/models/access_log.dart
class AccessLog {
  final String id;          // UUID
  final String userId;
  final String reservationId;
  final DateTime? checkInTime;
  final DateTime? checkOutTime;
  final String? deviceId;

  AccessLog({
    required this.id,
    required this.userId,
    required this.reservationId,
    this.checkInTime,
    this.checkOutTime,
    this.deviceId,
  });

  factory AccessLog.fromJson(Map<String, dynamic> json) {
    return AccessLog(
      id: json['log_id'],
      userId: json['user_id'],
      reservationId: json['reservation_id'],
      checkInTime: json['check_in_time'] != null ? DateTime.parse(json['check_in_time']) : null,
      checkOutTime: json['check_out_time'] != null ? DateTime.parse(json['check_out_time']) : null,
      deviceId: json['device_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'log_id': id,
      'user_id': userId,
      'reservation_id': reservationId,
      'check_in_time': checkInTime?.toIso8601String(),
      'check_out_time': checkOutTime?.toIso8601String(),
      'device_id': deviceId,
    };
  }
}

File: backend/src/main/java/slib/com/example/entity/hce/AccessLog.java
package slib.com.example.entity.hce; 

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;
import slib.com.example.entity.users.User;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "access_logs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class AccessLog {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID) 
    @Column(name = "log_id")
    private UUID logId;


    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @ToString.Exclude      
    @EqualsAndHashCode.Exclude
    private User user;

    @Column(name = "user_id", insertable = false, updatable = false)
    private UUID userId; 

    @Column(name = "reservation_id")
    private UUID reservationId;


    @Column(name = "check_in_time", nullable = false, updatable = false)
    @CreationTimestamp
    private LocalDateTime checkInTime;

    @Column(name = "check_out_time")
    private LocalDateTime checkOutTime; 

    @Column(name = "device_id", nullable = false, length = 50)
    private String deviceId; 
}